packages/events/src/util.ts

```diff
@@ -55,3 +56,47 @@ export function getEventServiceUrl(
     }
   });
 }
+
+/**
+ * Get the events query based on type.
+ * @param type - string to indicate event type with the options `upcoming`, `past`, `cancelled` and `draft`
+ * @param params - query features request options
+ * @returns an IQueryFeaturesRequestOptions that has the same values as `params` but for the modified Where and OrderBy properties
+ */
+export function getEventQueryFromType(
+  type: string,
+  params: IQueryFeaturesRequestOptions
```

Comment:
great work.

one last suggestion, change the name of `params` to `options`. I know it's only internal to this function, but it will appear in the docs, and in hub.js/rest-js land, "options" and "params" have specific meaning.